### PR TITLE
Added new apt_reboot_required fact, updated readme, and added unit tests

### PR DESCRIPTION
**Pull request by dlactin**
Simple fact to determine if a reboot is necessary after updates have been installed, along with unit tests and comment in the README.

### CODE REVIEW

**Comment by mhaskel** [comment]
Hi @dlactin, sorry for the inconveniencce, but this will need a rebase since the README update was just merged. Let me know if you need any help with that!

**Comment by dlactin** [comment]
I think I am all set, let me know if you need me to to something else.

File: README.md [hunk 1]
```diff
@@ -443,4 +445,4 @@ Puppet Labs modules on the Puppet Forge are open projects, and community contrib
 
 For more information, see our [module contribution guide.](https://docs.puppetlabs.com/forge/contributing.html)
 
-To see who's already involved, see the [list of contributors.](https://github.com/puppetlabs/puppetlabs-apt/graphs/contributors)
```
Comment (mhaskel) [inline README.md]: This line shouldn't have been removed

**Comment by mhaskel** [comment]
@dlactin looks like one line was accidentally removed from the README in the rebase, but once you add that back in we're good to go! thanks!

**Comment by dlactin** [comment]
I forgot to turn off eol before I finished my changes. Should be good to go!

**Comment by mhaskel** [comment]
:+1: thanks @dlactin !

**Pull request merged**